server/service/service_queries.go

```diff
@@ -5,8 +5,38 @@ import (
 
 	"github.com/kolide/fleet/server/contexts/viewer"
 	"github.com/kolide/fleet/server/kolide"
+	"github.com/pkg/errors"
 )
 
+func (svc service) ApplyQueryYaml(ctx context.Context, yml string) error {
```

Comment:
The implementation here is just serialization and could be solved in `encodeResponse`/`encodeRequest` methods no?

Reply:
IMO yaml is not a transport serialization format and should not be handled in the transport (en/de)coding.

Reply:
I agree with @groob, if we follow the design enumerated in https://github.com/kolide/fleet/issues/1667, the client will never actually send the server the full yaml document.